### ISSUE TITLE
Low success rate when reproducing official BC baseline on PickCube-v1 with state motionplanning demos

### ISSUE THREAD

**Issue by DaiShuHeng** [haosulab/ManiSkill#1400]
Hi, I am trying to reproduce the official behavior cloning (BC) baseline on ManiSkill for PickCube-v1, but I am seeing very low success rates even when using the official BC script and the state-based motionplanning demos.

## Environment
- ManiSkill version: [fill in]
- Commit / branch: [fill in]
- Python version: 3.10
- PyTorch version: [fill in]
- OS: Ubuntu [fill in]

### 1. Generated the state demo file
I first generated the state-based demo file from the motionplanning demos, and then used that file for BC training.

Demo file used:
`/home/daishuheng/maniskill_project/demos/PickCube-v1/motionplanning/demos/PickCube-v1/motionplanning/trajectory.state.pd_ee_delta_pos.physx_cpu.h5`

### 2. Trained the official BC baseline
I used the official `examples/baselines/bc/bc.py` script with the following command:

```bash
python bc.py \
  --env-id "PickCube-v1" \
  --demo-path /home/daishuheng/maniskill_project/demos/PickCube-v1/motionplanning/demos/PickCube-v1/motionplanning/trajectory.state.pd_ee_delta_pos.physx_cpu.h5 \
  --control-mode "pd_ee_delta_pos" \
  --sim-backend "cpu" \
  --num-eval-envs 1 \
  --max-episode-steps 100 \
  --total-iters 10000

Training behavior

The training loss decreases normally, so optimization itself seems to work:

Iteration 0: loss ≈ 0.2615

Iteration 1000: loss ≈ 0.0058

Iteration 2000: loss ≈ 0.0040

Iteration 3000: loss ≈ 0.0017

Iteration 4000: loss ≈ 0.0015

Iteration 5000: loss ≈ 0.0018

Iteration 6000: loss ≈ 0.0013

Iteration 7000: loss ≈ 0.00094

Iteration 8000: loss ≈ 0.00065

Iteration 9000: loss ≈ 0.00017

However, the task success rate remains very low.

Best metrics observed during training were approximately:

success_once = 0.05

success_at_end = 0.04

So the model seems to fit the demonstrations, but closed-loop task performance is still poor.

Checkpoint reevaluation

I modified bc.py locally to support loading a saved checkpoint and evaluating it separately. Then I reevaluated the best saved checkpoint with:python bc.py \
  --env-id "PickCube-v1" \
  --demo-path /home/daishuheng/maniskill_project/demos/PickCube-v1/motionplanning/demos/PickCube-v1/motionplanning/trajectory.state.pd_ee_delta_pos.physx_cpu.h5 \
  --control-mode "pd_ee_delta_pos" \
  --sim-backend "cpu" \
  --num-eval-envs 1 \
  --num-eval-episodes 5 \
  --max-episode-steps 100 \
  --checkpoint runs/PickCube-v1__bc__1__1774277476/checkpoints/best_eval_success_at_end.pt \
  --evaluate-only \
  --exp-name eval_best_at_end_h100_debug
Reevaluation result:

success_once: 0.0000

success_at_end: 0.0000

episode_len: 100.0000

What I observed from the evaluation videos

From the saved videos, the failures mainly look like this:

the gripper fails to grasp the red cube at all, or

the gripper grasps the cube but fails to place it at the correct target position

So the policy does not look completely random, but it is clearly not robust enough to solve the task reliably.

My question

Is this level of performance expected for the official BC baseline on PickCube-v1 with state-based motionplanning demos and pd_ee_delta_pos control?

More specifically:

Is BC known to perform poorly on this task, even with the official setup?

Am I missing any important settings or implementation details for reproducing this baseline?

Would you recommend trying --normalize-states for this setup?

Is it expected that ACT or Diffusion Policy should perform much better than BC here?

Additional note

At first I accidentally reevaluated without setting --max-episode-steps 100, which resulted in episode_len = 50 and truncated videos. After fixing that and reevaluating with episode_len = 100, the policy still had near-zero success.

Any guidance would be greatly appreciated. Thanks.

**Comment by StoneT2000** [comment]
That is expected actually. PickCube is a pretty difficult problem (relative to other benchmarks' pick cube style tasks like robosuite) due to the amount of randomization in the cube pose and the goal position.

ACT and diffusion policy should perform much better, maybe near 100% success rate.
